geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/QuadTreeFeatureStore.scala

```diff
@@ -0,0 +1,44 @@
+package org.locationtech.geomesa.utils.index
+
+import com.vividsolutions.jts.geom.Geometry
+import org.geotools.data.FeatureReader
+import org.geotools.data.collection.DelegateFeatureReader
+import org.geotools.data.store.FilteringFeatureIterator
+import org.geotools.feature.collection.DelegateFeatureIterator
+import org.geotools.geometry.jts.JTS
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.expression.{Literal, PropertyName}
+import org.opengis.filter.spatial.{BBOX, BinarySpatialOperator, Within}
+
+trait QuadTreeFeatureStore {
+  type FR = FeatureReader[SimpleFeatureType, SimpleFeature]
+  type DFR = DelegateFeatureReader[SimpleFeatureType, SimpleFeature]
+  type DFI = DelegateFeatureIterator[SimpleFeature]
+
+  def qt: SynchronizedQuadtree
+  def sft: SimpleFeatureType
+
+  def within(w: Within): FR = {
+    val (_, geomLit) = splitBinOp(w)
+    val geom = geomLit.evaluate(null).asInstanceOf[Geometry]
+    val res = qt.query(geom.getEnvelopeInternal)
+    val fiter = new DFI(res.asInstanceOf[java.util.List[SimpleFeature]].iterator)
+    val filt = new FilteringFeatureIterator[SimpleFeature](fiter, w)
```

Comment:
(Picky) Given that there is a filter and an iterator of features, fiter might not be the best name;)